File: dotnet/DesignPatterns/StructuralDesignPatterns/BridgeDesignPattern/BridgeDesignPattern.Example4/Bike.cs
namespace BridgeDesignPattern.Example4;

public class Bike : Vehicle
{
	public Bike(Workshop workshop1, Workshop workshop2) : base(workshop1, workshop2)
	{
	}

	public override void Manufacture()
	{
		Console.Write("Bike");
		_workShop1.Work();
		_workShop2.Work();
	}
}

File: nodejs/DesignPatterns/StructuralDesignPatterns/BridgeDesignPattern/BridgeDesignPattern.Example4/src/Bike.ts
import { IWorkshop } from './IWorkshop';
import { Vehicle } from './Vehicle';

export class Bike extends Vehicle {
  constructor(w1: IWorkshop, w2: IWorkshop) { super(w1, w2); }
  manufacture(): void {
    console.log(`Bike${this.workshop1.work()}${this.workshop2.work()}`);
  }
}
